lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -111,6 +112,7 @@ pub struct ShardReplicaSet {
     write_ordering_lock: Mutex<()>,
     /// Local clock set, used to tag new operations on this shard.
     clock_set: Mutex<ClockSet>,
+    write_rate_limiter: parking_lot::Mutex<Option<RateLimiter>>,
```

Comment:
Shouldn't this be an `Option<Mutex<RateLimiter>>`? For the regular case of no limiting, this might introduce overhead when locking

Reply:
Thanks for the idea :+1: 

I applied it in https://github.com/qdrant/qdrant/pull/5653/commits/a5cbe8f6c6f16c8d00c8614f69d257c17db6df73 and even found a bug regarding disabling rate limiting on the way!

If you are happy with this I will apply the same transformation and fix to the read limiter in a follow up PR

Reply:
Yes, this looks good!

I like that we now prevent locking if disabled. And the mutable self references we now need to update the rate limiter seem fine to me.

Thanks for adding support for disabling as well.